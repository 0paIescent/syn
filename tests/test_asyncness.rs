extern crate syn;

mod features;

#[macro_use]
mod macros;

use syn::{Expr, Item};

#[test]
fn test_async_fn() {
    let code = "async fn process() {}";
    snapshot!(code as Item);
}

#[test]
fn test_async_closure() {
    let code = "async || {}";
    snapshot!(code as Expr);
}
