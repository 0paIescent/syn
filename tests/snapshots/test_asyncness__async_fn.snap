---
created: "2019-03-11T05:39:50.464009531Z"
creator: insta@0.7.1
source: tests/test_asyncness.rs
expression: syntax_tree
---
Fn(
    ItemFn {
        attrs: [],
        vis: Inherited,
        constness: None,
        unsafety: None,
        asyncness: Some(
            Async
        ),
        abi: None,
        ident: Ident(
            process
        ),
        decl: FnDecl {
            fn_token: Fn,
            generics: Generics {
                lt_token: None,
                params: [],
                gt_token: None,
                where_clause: None
            },
            paren_token: Paren,
            inputs: [],
            variadic: None,
            output: Default
        },
        block: Block {
            brace_token: Brace,
            stmts: []
        }
    }
)
